lib/resources/postgres.rb

```diff
@@ -6,22 +6,30 @@
 
 module Inspec::Resources
   class Postgres < Inspec.resource(1)
-    name 'postgres'
-
-    attr_reader :service, :data_dir, :conf_dir, :conf_path
+  name 'postgres'
+  attr_reader :service, :data_dir, :conf_dir, :conf_path, :version
     def initialize
+      data_dir_command = conf_dir_from_process
+      version_command = version_from_process(data_dir_command)
       os = inspec.os
       if os.debian?
-        #
         # https://wiki.debian.org/PostgreSql
         #
         # Debian allows multiple versions of postgresql to be
         # installed as well as multiple "clusters" to be configured.
         #
-        version = version_from_dir('/etc/postgresql')
+        if !data_dir_command.nil?
```

Comment:
This is a lot of logic for an initialize method. We should really break this out to a helper method so this can become:

```
@data_dir = detect_data_dir
```

... since the logic is used in multiple places within initialize.